### PR TITLE
docs(applicationset): correct typos in documentation

### PR DESCRIPTION
**Pull request by SuminSSon**
### Description

This PR corrects three typos across various ApplicationSet documentation files.

- Changed "duplicated keys" to "duplicate keys" in configurations.md.
- Corrected "GitPod" to "Gitpod" in setup.md.
- Updated "cluster1" to "cluster2" in cluster.md example.

### Related Issue

Closes [ISSUE [#20553](https://github.com/argoproj/argo-cd/issues/20553)]

### Reason for Change

These corrections enhance clarity and accuracy, reducing potential confusion for readers and improving overall document quality.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by nitishfy — CHANGES_REQUESTED** [review]
Overall LGTM! 

Can you go ahead and fix the Lint issues and sign-off the commits?

**Comment by SuminSSon** [comment]
I've addressed the lint issues and ensured all commits are signed off as requested. Let me know if there are any further adjustments you'd like me to make. Thanks!

File: cmd/argocd/commands/admin/app.go [hunk 1]
```diff
@@ -188,12 +188,12 @@ func NewDiffReconcileResults() *cobra.Command {
 func toUnstructured(val interface{}) (*unstructured.Unstructured, error) {
 	data, err := json.Marshal(val)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error while marhsalling value: %w", err)
```
Comment (rumstead) [inline cmd/argocd/commands/admin/app.go:191]:
```suggestion
		return nil, fmt.Errorf("error while marshaling value: %w", err)
```

File: cmd/argocd/commands/admin/app.go [hunk 2]
```diff
@@ -188,12 +188,12 @@ func NewDiffReconcileResults() *cobra.Command {
...
 	}
 	res := make(map[string]interface{})
 	err = json.Unmarshal(data, &res)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error while unmarhsalling data: %w", err)
```
Comment (rumstead) [inline cmd/argocd/commands/admin/app.go:196]:
```suggestion
		return nil, fmt.Errorf("error while unmarshalling data: %w", err)
```

**Comment by rumstead** [comment]
I left two NIT comments on spelling. Up to you if you want to use the British or US spelling for marshaling.

**Pull request closed without merging**